### PR TITLE
feat: Add notifications to Menu

### PR DESCRIPTION
**Pull request by cjakeman**
Blueprint at [https://blueprints.launchpad.net/or/+spec/update-notifications](https://blueprints.launchpad.net/or/+spec/update-notifications)

Forum thread at [http://www.elvastower.com/forums/index.php?/topic/35837-notifications](http://www.elvastower.com/forums/index.php?/topic/35837-notifications).
This shows use cases which present additional information about a code update or about content.

- [X] Add support for updating Open Rails
- [X] Document the creating, testing and publishing of new notifications
- [x] Extend to multiple notifications
- [x] Activate the indicator for new notifications
- [x] Fetch notifications in a background task, so done only once.
- [x] Check fetch from remote site fails gracefully
- [x] Check position of items on form for all Windows resolutions
- [x] Update Manual
- [x] Redirect to fetch data from real source not test source

### CODE REVIEW

**Comment by twpol** [comment]
FYI looks like the update label is overlapping (behind) the what's new label:

image

**Comment by cjakeman** [comment]
I seem to have broken the Update mechanism, so I've marked this PR as not-for-unstable until I can fix it.

Sorry.

**Comment by cjakeman** [comment]
The Word document Source/Documentation/Notifications.docx shows examples of use with JSON files, the presentation in Menu.exe and also the debugging tools.

File: Source/Menu/Notifications/NotificationManager.cs [hunk 1]
```diff
@@ -0,0 +1,674 @@
+﻿// COPYRIGHT 2009, 2010, 2011, 2012, 2013, 2014, 2015 by the Open Rails project.
```
Comment (twpol) [inline Source/Menu/Notifications/NotificationManager.cs:1]: Probably could update the years here :) I've used a range for newer files as it's simpler.

File: Source/Menu/Notifications/Notifications.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    //public class Met
+    //{
+    //    public List<Item> ItemList { get; set; }
+    //    public List<CheckId> CheckIdList { get; set; }
+    //}
+    //public class CheckId
+    //{
+    //    public string Id { get; set; }
+    //}
```
Comment (twpol) [inline Source/Menu/Notifications/Notifications.cs:84]: Should we remove these bits of commented-out code?

File: Source/ORTS.Common/SystemInfo.cs [hunk 3]
```diff
@@ -67,7 +67,9 @@ static SystemInfo()
             }
             catch (ManagementException error)
             {
-                Trace.WriteLine(error);
+                // Likely to catch multiple exceptions like:
+                // Exception thrown: 'System.IO.InvalidDataException' in ORTS.Menu.dll
+                //Trace.WriteLine(error);
```
Comment (twpol) [inline Source/ORTS.Common/SystemInfo.cs:72]: Did we have to remove the trace? These errors are useful diagnostics in the log file

**Review by twpol — APPROVED** [review]
Approved with some comments that can be dealt with in this PR or a subsequent at your choosing

File: Source/Menu/Notifications/NotificationManager.cs (hunk 1 above)
Comment (cjakeman) [reply Source/Menu/Notifications/NotificationManager.cs]: Good idea. I read something a while back on the copyright dates and [this looks interesting](https://liferay.dev/blogs/-/blogs/how-and-why-to-properly-write-copyright-statements-in-your-code)

File: Source/Menu/Notifications/Notifications.cs (hunk 2 above)
Comment (cjakeman) [reply Source/Menu/Notifications/Notifications.cs]: Thanks for spotting this.

File: Source/ORTS.Common/SystemInfo.cs (hunk 3 above)
Comment (cjakeman) [reply Source/ORTS.Common/SystemInfo.cs]: Restored

**Comment by cjakeman** [comment]
To play with the notifications, then try these files attached and save them in the Program folder.
[notifications_trial.json](https://github.com/user-attachments/files/17062363/notifications_trial.json)
[notifications_trial_parameters.json](https://github.com/user-attachments/files/17062366/notifications_trial_parameters.json)

File: Source/Menu/MainForm.cs [hunk 4]
```diff
@@ -1613,6 +1613,15 @@ public void UpdateNotificationPageAlert()
...
                 lblNotificationCount.Text = $"{NotificationManager.NewPages.Count - NotificationManager.NewPages.Viewed}";
+
+                // If screen scaling != 100%, then the count doesn't lie on top of the red circle, so adjust its position.
+                if (NotificationManager.ScreenScaling != 1.0 & NotificationManager.ScreenAdjusted == false)
+                {
+                    var adjustment = (int)((NotificationManager.ScreenScaling - 1.0) * 20); // 20 to adjust 125% by 5 pixels
+                    lblNotificationCount.Top -= adjustment;
+                    lblNotificationCount.Left -= adjustment;
+                    NotificationManager.ScreenAdjusted = true;
+                }
```
Comment (twpol) [inline Source/Menu/MainForm.cs:1624]: Fun! We should probably follow up with a look at https://learn.microsoft.com/en-us/dotnet/desktop/winforms/high-dpi-support-in-windows-forms which may improve scaling

**Review by twpol — APPROVED** [review]

**Pull request merged**